packages/medusa/src/migrations/1656949291839-sales_channel.ts

```diff
@@ -0,0 +1,75 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export const featureFlag = "sales-channels"
+
+export class salesChannel1656949291839 implements MigrationInterface {
+  name = "salesChannel1656949291839"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE TABLE "sales_channel" ("id" character varying NOT NULL, "created_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "deleted_at" TIMESTAMP WITH TIME ZONE, "name" character varying NOT NULL, "description" character varying, "disabled_at" TIMESTAMP WITH TIME ZONE, CONSTRAINT "PK_d1eb0b923ea5a0eb1e0916191f1" PRIMARY KEY ("id"))`
+    )
+    await queryRunner.query(
+      `CREATE TABLE "product_sales_channel" ("product_id" character varying NOT NULL, "sales_channel_id" character varying NOT NULL, CONSTRAINT "PK_fd29b6a8bd641052628dee19583" PRIMARY KEY ("product_id", "sales_channel_id"))`
+    )
+
+    await queryRunner.query(
+      `CREATE INDEX "IDX_5a4d5e1e60f97633547821ec8d" ON "product_sales_channel" ("product_id") `
+    )
+    await queryRunner.query(
+      `CREATE INDEX "IDX_37341bad297fe5cca91f921032" ON "product_sales_channel" ("sales_channel_id") `
+    )
+
+    await queryRunner.query(
+      `ALTER TABLE "public"."cart" ADD "sales_channel_id" character varying NOT NULL`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "public"."order" ADD "sales_channel_id" character varying NOT NULL`
+    )
+
```

Comment:
In order to support development while this is not being deployed and run we should consider removing the `NOT NULL` constraint. This constraint will cause the migration to fail when it's run against an existing database with carts and orders. 

I suggest we remove them now. When we in the future should support migrating existing databases to sales channels, we could: 
-  generate an id
- insert a sales channel with the id 
- set the id as default value for the columns

Reply:
https://github.com/medusajs/medusa/pull/1783

Reply:
Would even argue that in the long run we should still allow null in `sales_channel_id`; at least on orders in case they are created from admin.

Reply:
That is a good point. What do you think we should do wrt. migrating the existing entities?

Reply:
It can be useful to display the date of when it has been disabled by the user, that gives an extra information to the user. But for now if we don’t want that info lets stick on the boolean

Reply:
Would actually just leave them as is - ie allow them to be "unchannelled" :)